### ISSUE TITLE
Fix `declaration-block-no-redundant-longhand-properties` false positives for `inherit` keyword

### ISSUE THREAD

**Issue by AlexElin** [stylelint/stylelint#6394]
### What steps are needed to reproduce the bug?

1. Run stylint for checking the below CSS

```css
.demo {
  border-top-left-radius: 20px;
  border-top-right-radius: inherit;
  border-bottom-right-radius: inherit;
  border-bottom-left-radius: inherit;
}
```

### What Stylelint configuration is needed to reproduce the bug?

```json
{
  "extends": "stylelint-config-standard"
}
```

### How did you run Stylelint?

stylelint src/**/*.css

### Which version of Stylelint are you using?

^14.13.0

### What did you expect to happen?

There's no error for this case because there are longhand properties with `inherit` value.
Per [MDN](https://developer.mozilla.org/en-US/docs/Web/CSS/Shorthand_properties#omitting_properties)

> Only the individual properties values can inherit. As missing values are replaced by their initial value, it is impossible to allow inheritance of individual properties by omitting them. The keyword inherit can be applied to a property, but only as a whole, not as a keyword for one value or another. That means that the only way to make some specific value to be inherited is to use the longhand property with the keyword inherit.

So, it's impossible to convert such set of longhand properties to shorthand and preserve initial semantic.

### What actually happened?

 5:3  ✖  Expected shorthand property "border-radius"  declaration-block-no-redundant-longhand-properties

### Proposal to fix the bug

If a longhand property of some shorthand property is set to `inherit`, don't treat such case as a rule violation.

**Comment by jeddy3** [comment]
@AlexElin Thanks for the report and for using the template.

I've labelled the issue as ready to implement. Please consider [contributing](https://stylelint.io/contributing) if you have time.

There are [steps on how to fix a bug in a rule](https://stylelint.io/developer-guide/rules#fix-a-bug-in-a-rule) in the Developer guide.
